.github/workflows/publish-packages.yml

```diff
@@ -323,5 +307,136 @@ jobs:
+        uses: actions/checkout@v2.3.2
+        with:
+          path: 'hazelcast-packaging'
+
+      - name: Cache Vagrant boxes
+        uses: actions/cache@v2
+        with:
+          path: ~/.vagrant.d/boxes
+          key: ${{ runner.os }}-vagrant-${{ hashFiles('Vagrantfile') }}
+          restore-keys: |
+            ${{ runner.os }}-vagrant-
+
+      - name: Show Vagrant version
+        run: vagrant --version
+
+      - name: Run Vagrant up to cache machines
+        run: |
+          source ./common.sh
+          vagrant up
+
+      - name: Run Vagrant halt
+        run: vagrant halt
+
+  testOnVagrant:
+    runs-on: macos-12
+    needs: [prepare, buildVagrant]
+
+    defaults:
+      run:
+        working-directory: ./hazelcast-packaging
+
+    continue-on-error: true
+
+    strategy:
+      fail-fast: false
+      matrix:
+        hz_distribution: [ 'hazelcast', 'hazelcast-enterprise' ]
+        linux_distribution: [ 'fedora', 'ubuntu' ]
+    env:
+      HZ_DISTRIBUTION: ${{ matrix.hz_distribution }}
+      LINUX_DISTRIBUTION: ${{ matrix.linux_distribution }}
+      VAGRANT_CWD: ${{ github.workspace }}/hazelcast-packaging/vagrant
+      HZ_VERSION: ${{ needs.prepare.outputs.hz_version }}
+      PACKAGE_VERSION: ${{ needs.prepare.outputs.package_version }}
+
+    steps:
+      - name: Checkout hazelcast-packaging repo
+        uses: actions/checkout@v2.3.2
+        with:
+          path: 'hazelcast-packaging'
+
+      - name: Cache Vagrant boxes
+        uses: actions/cache@v2
+        with:
+          path: ~/.vagrant.d/boxes
+          key: ${{ runner.os }}-vagrant-${{ hashFiles('Vagrantfile') }}
+          restore-keys: |
+            ${{ runner.os }}-vagrant-
+
+      - name: Run Vagrant machine ${{ env.LINUX_DISTRIBUTION}}
+        run: |
+          source ./common.sh
+          vagrant up ${{ env.LINUX_DISTRIBUTION}}
+
+      - name: Run Hazelcast ${{ matrix.hz_distribution }} on ${{ matrix.linux_distribution }}
+        run: |
+          vagrant ssh ${{ env.LINUX_DISTRIBUTION}} -c \
+            'HAZELCAST_CONFIG="/home/vagrant/hz-config.yaml" hz-start > ~/hz.log 2>&1 &'
+
+      - name: Wait for cluster to be up
+        run: |
+          vagrant ssh ${{ env.LINUX_DISTRIBUTION}} -c '~/check-hazelcast-health.sh'
+
+      - name: check health via script
+        run: vagrant ssh ${{ env.LINUX_DISTRIBUTION}} -c 'hz-healthcheck'
+
+      - name: stop machine
+        run: vagrant halt ${{ env.LINUX_DISTRIBUTION}}
+
+  cleanup:
+    runs-on: ubuntu-latest
+    if: always()
+    container: rockylinux:9
+    strategy:
+      fail-fast: false
+      matrix:
+        distribution: [ 'hazelcast', 'hazelcast-enterprise' ]
+    env:
+      HZ_VERSION: ${{ needs.prepare.outputs.hz_version }}
+      PACKAGE_VERSION: ${{ needs.prepare.outputs.package_version }}
+      HZ_DISTRIBUTION: ${{ matrix.distribution }}
+    defaults:
+      run:
+        working-directory: ./hazelcast-packaging
+    needs: [ prepare, testOnVagrant ]
```

Comment:
Same here about `prepare`

Reply:
For the `HZ_VERSION: ${{ needs.prepare.outputs.hz_version }}` as everywhere